code/controllers/subsystem/mapping.dm

```diff
@@ -251,7 +252,10 @@ SUBSYSTEM_DEF(mapping)
 	// Generate deep space ruins
 	var/list/space_ruins = levels_by_trait(ZTRAIT_SPACE_RUINS)
 	if (space_ruins.len)
-		seedRuins(space_ruins, CONFIG_GET(number/space_budget), list(/area/space), themed_ruins[ZTRAIT_SPACE_RUINS])
+		// Create a proportional budget by multiplying the amount of space ruin levels in the current map over the default amount
+		var/default_ruin_amount = 7
+		var/proportional_budget = CONFIG_GET(number/space_budget) * round(space_ruins.len / default_ruin_amount)
```

Comment:
im pretty sure that round() is incorrect, did you test this?

Reply:
i was trying to figure that out now, it worked before but it was like Ruin loader finished with 0.256452 left to spend so i tried to fix it with round.

Reply:
oh i finally figrued it out after 1 hour, if you round a fraction it just makes it one bruh